triggers/call_notification.js

```diff
@@ -0,0 +1,177 @@
+    },
+    body: JSON.stringify(data),
+  };
+
+  // You may return a promise or a normal data structure from any perform method.
+  return z.request(options)
+    .then(response => JSON.parse(response.content));
+};
+
+const unsubscribeHook = (z, bundle) => {
+  // bundle.subscribeData contains the parsed response JSON from the subscribe
+  // request made initially.
+  const hookId = bundle.subscribeData.id;
+
+  // You can build requests and our client will helpfully inject all the variables
+  // you need to complete. You can also register middleware to control this.
+  const options = {
+    url: `${process.env.PLATFORM_URL}/api/callnotification/callnotification/${hookId}`,
+    method: 'DELETE',
+    headers: {
+      Authorization: `Token ${bundle.authData.email}:${bundle.authData.token}`,
+    }
+  };
+
+  // You may return a promise or a normal data structure from any perform method.
+  return z.request(options)
+    .then(response => z.console.log(response));
+};
+
+const getCallNotification = (z, bundle) => [callNotification];
+
+module.exports = {
+  key: 'call_notification',
+  noun: 'Call',
+
+  display: {
+    label: 'New Call Notification',
+    description: 'Triggers when a new call notification occurs.',
+    hidden: false,
+    important: false,
+  },
+
+  operation: {
+    type: 'hook',
+    performSubscribe: subscribeHook,
+    performUnsubscribe: unsubscribeHook,
+
+    inputFields: [
+      {
+        key: 'call_notification',
+        label: 'Call Notification',
+        type: 'string',
+        required: false,
+      },
+    ],
+    outputFields: [
+      {
+        key: 'id',
+        type: 'string',
+      },
+      {
+        key: 'call_id',
+        type: 'string',
+      },
+      {
+        key: 'caller__account_number',
+        type: 'string',
+      },
+      {
+        key: 'caller__name',
+        type: 'string',
+      },
+      {
+        key: 'caller__number',
+        type: 'string',
+      },
+      {
+        key: 'caller__user_numbers',
+        type: 'string',
+      },
+      {
+        key: 'destination__number',
+        type: 'string',
+      },
+      {
+        key: 'destination__targets',
+        type: 'string',
+      },
+      {
+        key: 'destination__targets[]account_number',
+        type: 'string',
+      },
+      {
+        key: 'destination__targets[]name',
+        type: 'string',
+      },
+      {
+        key: 'destination__targets[]number',
+        type: 'string',
+      },
+      {
+        key: 'destination__targets[]user_numbers',
+        type: 'string',
+      },
+      {
+        key: 'direction',
+        type: 'string',
+      },
+      {
+        key: 'status',
+        type: 'string',
+      },
+      {
+        key: 'timestamp',
+        type: 'string',
+      },
+      {
+        key: 'version',
+        type: 'string',
+      },
+    ],
+    perform: getCallNotification,
+    performList: getCallNotification,
+    sample: callNotification,
```

Comment:
The name `callNotification` is a bit confusing, I would suggest renaming it to `sampleCallNotification`, or just embed the whole thing in here.

Reply:
The `callNotification` can now actually return a real call notification object. 
I have added a `getFallbackCallNotification` because that one is used when the polling of the hook returns nothing, so IMO that would be a fine name.